source/apps/crawl_space/viz/stream.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import division
+from __future__ import print_function
+
+from datetime import datetime
+import json
+from Queue import Empty
+
+import numpy as np
+from bokeh.plotting import figure, output_server
+from kombu import Connection
+
+from bokeh.models.glyphs import Segment
+from bokeh.models.markers import Circle
+from bokeh.models import Range1d, LinearAxis, ColumnDataSource
+from bokeh.embed import autoload_server
+from bokeh.session import Session
+from bokeh.document import Document
+
+DEFAULT_NUM_URLS=25
+
+
+def init_plot(name='fetcher_log'):
+    session = Session()
+    document = Document()
+    session.use_doc(name)
+    session.load_document(document)
+
+    if document.context.children:
+        plot = document.context.children[0]
+    else:
+        output_server(name)
+        # TODO: Remove these when Bokeh is upgraded
+        # placeholders or Bokeh can't inject properly
+        current = np.datetime64(datetime.now())
+        xdr = Range1d(current, current + 1)
+        ydr = ["urls"]
+
+        # styling suggested by Bryan
+        plot = figure(title="Crawler Monitor", tools="pan,wheel_zoom,resize,save,hover",
+                      x_axis_type="datetime", y_axis_location="right", x_range=xdr, y_range=ydr,
+                      width=1200, height=600)
+        plot.toolbar_location = None
+        plot.xgrid.grid_line_color = None
+
+    document.add(plot)
+    session.store_document(document)
+    script = autoload_server(plot, session)
+
+    #TODO: Looks like a Bokeh bug, probably not repeatable with current code
+    script = script.replace("'modelid': u'", "'modelid': '")
+    return script
+
+
+class NutchUrlTrails:
+    """
+    Class for managing URL Trails visualizations
+    """
+
+    @staticmethod
+    def strip_url(url):
+        """
+        Make a URL safe for visualization in Bokeh server
+        :param url: a URL to be shortened/stripped
+        :return: The stripped URL
+        """
+        # TODO: remove protocol-stripping on next Bokeh release
+        return url.replace('https://', '').replace('http://', '').replace(':', '_').replace('-', '_')[:250]
```

Comment:
Okay, let's try 50 characters and see how that goes.  I'll bring it out as a variable up top, and maybe we should even make it a setting.